sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-factory.ts

```diff
@@ -1,48 +1,61 @@
-import { MSFAANumber } from "@sims/sims-db";
+import { MSFAANumber, Student } from "@sims/sims-db";
 import {
   E2EDataSources,
+  MSFAAStates,
   createFakeMSFAANumber,
   createFakeStudent,
   createFakeUser,
   saveFakeApplication,
   saveFakeStudent,
 } from "@sims/test-utils";
-import { MSFAATestInputData } from "./msfaa-part-time-process-integration.scheduler.models";
+import { MSFAATestInputData } from "./msfaa-process-integration.scheduler.models";
 import { createFakeSINValidation } from "@sims/test-utils/factories/sin-validation";
 
 /**
  * Save an MSFAA record providing all data that will be
  * part of the MSFAA file generation.
  * @param db data source helper.
- * @param msfaa test input data.
+ * @param msfaaDataInput msfaa test input data.
+ * @param relations
+ * - `fakeStudent` fakeStudent to be associated with the MSFAATestInputData.
+ * @param options to saveMsfaaTestInputData.
+ * - `msfaaState` if provided, specifies the MSFAA State to be associated with the provided MSFAATestInputData.
+ * - `offeringIntensity` if provided, specifies the offering intensity associated with the provided MSFAATestInputData.
  * @returns a saved MSFAA record that uses the input test
  * data to be created.
  */
 export async function saveMSFAATestInputData(
   db: E2EDataSources,
   msfaaDataInput: MSFAATestInputData,
+  relations?: { fakeStudent?: Student },
+  options?: { msfaaState?: MSFAAStates },
 ): Promise<MSFAANumber> {
   // User.
   const fakeUser = createFakeUser();
   fakeUser.firstName = msfaaDataInput.givenName;
   fakeUser.lastName = msfaaDataInput.surname;
   fakeUser.email = msfaaDataInput.email;
   // Student.
-  const fakeStudent = createFakeStudent(fakeUser);
-  fakeStudent.birthDate = msfaaDataInput.birthDate;
-  fakeStudent.gender = msfaaDataInput.gender;
-  fakeStudent.contactInfo = {
-    address: {
-      addressLine1: msfaaDataInput.addressLine1,
-      addressLine2: msfaaDataInput.addressLine2,
-      city: msfaaDataInput.city,
-      country: msfaaDataInput.country,
-      selectedCountry: msfaaDataInput.country,
-      provinceState: msfaaDataInput.provinceState,
-      postalCode: msfaaDataInput.postalCode,
-    },
-    phone: msfaaDataInput.phone,
-  };
+  let fakeStudent: Student;
+  if (!relations?.fakeStudent) {
+    fakeStudent = relations?.fakeStudent ?? createFakeStudent(fakeUser);
```

Comment:
the relations?.fakeStudent ?? is redundant here as it is already validated for false at line 41.

Overall it doesn't matter as it is better to remove the student option in the method as per https://github.com/bcgov/SIMS/pull/2010/files#r1238828745